create_infotexts.py

```diff
@@ -114,6 +117,111 @@ def create_smvk_mm_link(item):
+
+    :param item: dictionary containing the metadata for an image.
+    :param kw_list: list of keywords found in column <Nyckelord>.
+    :return: string representing concatenation of old <Beskrivning> plus keywords list.
+    """
+    newdesc = description_str + "\n"
+    newdesc += "'''Nyckelord:'''\n"
+
+    for kw in kw_list:
+        newdesc += kw + "\n"
+    return newdesc
+
+
+def process_keyword_addition_to_description(description_str, keywords_list):
+    """
+    Append keywords to description, unless the only keyword is "Svenska Cypernexpeditionen". 
+    
+    :param fotonr: string representing the name of the image minus extension.
+    :param description_str: string respresenting the value of column <Beskrivning>.
+    :param keywords_str: string representing a comma-separated list of keywords.
+    :return: None (The altered description strings are returned in append_keywords_to_desc())
+    """
+    if keywords_list:
+        for kw in keywords_list:
+            if not kw.lower() in description_str.lower():
+                return  append_keywords_to_desc(description_str, keywords_list)
+            else:
+                return description_str
+    else:
+        return description_str
+
+def remove_svenska_cypernexpedition_from_description(description):
+    """
+    Remove the string "Svenska Cypernexpeditionen" from description, since it's in all descriptions.
+    
+    :param description: string containg the field value for column <Beskrivning> plus potential enrichments.
+    :return: String with "Svenska Cypernexpeditionen" removed.
+    """
+    new_string = re.sub(" Svenska Cypernexpeditionen\.?", "", description)
+
+    return new_string
+
+def process_addition_of_region_to_description(description_str, region_str, country_str):
+    """
+    Add <Region, foto> to description string, except when it is already present, with some smartness.
+    
+    :param description_str: String representing the processed/enriched description incl. <Nyckelord>.
+    :param region_str: String with the field value from column <Region ,foto> in metadata.
+    :param country_str: String with the field value from column <Land> in metadata. 
+    :return: String with possibly enriched description.
+    """
+    newdesc = description_str
+    if region_str != "":
+        newdesc += "\n'''Region:'''\n\n" + region_str
+        if country_str != "":
+            newdesc += ", " + country_str
+        newdesc += "\n"
+
+    return newdesc
+
+def enrich_description(item):
+    """
+    Try to add keywords and regional information to description.
+    
+    :param item: dictionary containing metadata for one image.
+    :return: string representing altered or unaltered description.
+    """
+    description = remove_svenska_cypernexpedition_from_description(item["Beskrivning"])
+
+    if not description.endswith("."):
+        description += "."
+
+    stripped_keywords = generate_list_of_stripped_keywords(item["Nyckelord"])
```

Comment:
If you are not going to use this here then I would pass `item["Nyckelord"]` to `process_keyword_addition_to_description()` instead and have that function call `generate_list_of_stripped_keywords()`